File: algoritmos/Questão 2/animal.php
<?php

class Animal{
	public $peso;
	public $qtde_patas;
	public function __construct($peso, $qtde_patas){
		$this->peso = $peso;
		$this->qtde_patas = $qtde_patas;
	}
	public function andar(){
		echo " e estou andando.";
	}
	public function comer(){
		echo " e estou comendo.";
	}
}
?>

File: algoritmos/Algoritmo 2/Algoritmo2/src/main/java/com/fabio/algoritmo2/Animal.java
package com.fabio.algoritmo2;

/**
 *
 * @author fabio
 */

public abstract class Animal {
    double peso;
    int n_patas;
    
    public Animal(double peso, int n_patas){
        this.peso = peso;
        this.n_patas = n_patas;
    }
    
    public void andar(){
        System.out.println("estou andando.\n");
    }
    
    public void comer(){
        System.out.println("estou comendo.\n");
    }

    public double getPeso() {
        return peso;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }

    public int getN_patas() {
        return n_patas;
    }

    public void setN_patas(int n_patas) {
        this.n_patas = n_patas;
    }

    @Override
    public String toString() {
        return "Animal{" + "peso=" + peso + ", n_patas=" + n_patas + '}';
    }
}
